Source/WebCore/platform/graphics/GraphicsLayerFactory.h

```diff
@@ -30,8 +30,7 @@
 
 namespace WebCore {
 
-class GraphicsLayerFactory {
-    WTF_MAKE_FAST_ALLOCATED;
+class GraphicsLayerFactory : public RefCounted<GraphicsLayerFactory>, public CanMakeWeakPtr<GraphicsLayerFactory> {
```

Comment:
Makes sense. I did this because directly making `RemoteLayerTreeContext` refcounted conflicted with `GraphicsLayerFactory` already being `WTF_MAKE_FAST_ALLOCATED`. Is moving `WTF_MAKE_FAST_ALLOCATED` to the child classes better?

Reply:
Please see this recent change from Geoff:
```
Added a compile time assertion that forbids classes that offer WeakPtr
    with no corresponding RefPtr or CheckedPtr
    https://bugs.webkit.org/show_bug.cgi?id=273569
    rdar://127296969
```

Looks related?

Reply:
Thanks Geoff, this explanation is helpful. I wound up just moving RefCounting/CanMakeWeakPtr from the client interface into the subclass and resolving the ambiguous lookup there.